peer/config.go

```diff
@@ -0,0 +1,228 @@
+
+	// SigPool is used when creating *lnwallet.LightningChannel instances.
+	SigPool *lnwallet.SigPool
+
+	// Wallet is used to publish transactions and generate delivery scripts
+	// during the coop close process.
+	Wallet *lnwallet.LightningWallet
+
+	// ChainNotifier is used to receive confirmations of a coop close
+	// transaction.
+	ChainNotifier chainntnfs.ChainNotifier
+
+	// RoutingPolicy is used to set the forwarding policy for links created by
+	// the Brontide.
+	RoutingPolicy htlcswitch.ForwardingPolicy
+
+	// Sphinx is used when setting up ChannelLinks so they can decode sphinx
+	// onion blobs.
+	Sphinx *hop.OnionProcessor
+
+	// WitnessBeacon is used when setting up ChannelLinks so they can add any
+	// preimages that they learn.
+	WitnessBeacon contractcourt.WitnessBeacon
+
+	// Invoices is passed to the ChannelLink on creation and handles all
+	// invoice-related logic.
+	Invoices *invoices.InvoiceRegistry
+
+	// ChannelNotifier is used by the link to notify other sub-systems about
+	// channel-related events and by the Brontide to subscribe to
+	// ActiveLinkEvents.
+	ChannelNotifier *channelnotifier.ChannelNotifier
+
+	// HtlcNotifier is used when creating a ChannelLink.
+	HtlcNotifier *htlcswitch.HtlcNotifier
+
+	// TowerClient is used when creating a ChannelLink.
+	TowerClient wtclient.Client
+
+	// DisconnectPeer is used to disconnect this peer if the cooperative close
+	// process fails.
+	DisconnectPeer func(*btcec.PublicKey) error
+
+	// GenNodeAnnouncement is used to send our node announcement to the remote
+	// on startup.
+	GenNodeAnnouncement func(bool,
+		...netann.NodeAnnModifier) (lnwire.NodeAnnouncement, error)
+
+	// PrunePersistentPeerConnection is used to remove all internal state
+	// related to this peer in the server.
+	PrunePersistentPeerConnection func([33]byte)
+
+	// FetchLastChanUpdate fetches our latest channel update for a target
+	// channel.
+	FetchLastChanUpdate func(lnwire.ShortChannelID) (*lnwire.ChannelUpdate,
+		error)
+
+	// ProcessFundingOpen is used to hand off an OpenChannel message to the
+	// funding manager.
+	ProcessFundingOpen func(*lnwire.OpenChannel, lnpeer.Peer)
+
+	// ProcessFundingAccept is used to hand off an AcceptChannel message to the
+	// funding manager.
+	ProcessFundingAccept func(*lnwire.AcceptChannel, lnpeer.Peer)
+
+	// ProcessFundingCreated is used to hand off a FundingCreated message to
+	// the funding manager.
+	ProcessFundingCreated func(*lnwire.FundingCreated, lnpeer.Peer)
+
+	// ProcessFundingSigned is used to hand off a FundingSigned message to the
+	// funding manager.
+	ProcessFundingSigned func(*lnwire.FundingSigned, lnpeer.Peer)
+
+	// ProcessFundingLocked is used to hand off a FundingLocked message to the
+	// funding manager.
+	ProcessFundingLocked func(*lnwire.FundingLocked, lnpeer.Peer)
+
+	// ProcessFundingError is used to hand off an Error message to the funding
+	// manager.
+	ProcessFundingError func(*lnwire.Error, *btcec.PublicKey)
```

Comment:
How about a handler for any unknown messages? Could eventually allow users to import `lnd` into another project then manually handle some new extension they're experimenting with.

Reply:
I think this should be deferred to a followup PR, this one was mostly meant to be a code move. I agree that there should be a handler for unknown messages though.